goal_src/jak2/pc/pckernel.gc

```diff
@@ -61,20 +62,22 @@
 
 (defmethod update-discord-rpc pc-settings-jak2 ((obj pc-settings-jak2))
   "update discord rpc module"
-  (let ((info (new 'stack 'discord-info)))
-    (set! (-> info orb-count) (&-> *game-info* skill-total))
-    (set! (-> info gem-count) (&-> *game-info* gem-total))
-    (set! (-> info death-count) (&-> *game-info* total-deaths))
-    (set! (-> info status) "Playing Jak II™")
-    ;; grab the name of the level we're in
-    (if (and *level* (level-get-target-inside *level*))
-      (set! (-> info level) (symbol->string (-> (level-get-target-inside *level*) name)))
-      (set! (-> info level) "unknown"))
-    (set! (-> info cutscene?) #f) ;; TODO - cutscenes don't work yet
-    (set! (-> info time-of-day) (&-> *time-of-day-context* time))
-    (set! (-> info percent-complete) (calculate-percentage *game-info*))
-    ;; TODO - wrapping in `with-profiler` causes an error, fix it
-    (pc-discord-rpc-update info))
+  (set! *discord-info* (new 'stack 'discord-info))
+  (set! (-> *discord-info* orb-count) (-> *game-info* skill-total))
+  (set! (-> *discord-info* gem-count) (-> *game-info* gem-total))
+  (set! (-> *discord-info* death-count) (-> *game-info* total-deaths))
+  (set! (-> *discord-info* task) "unknown")
+  (set! (-> *discord-info* status) (the string (get-active-mission-description)))
```

Comment:
Don't cast here, it's weird.

Reply:
I could have sworn I got an error about the function returning `object` at some point which is why I did that cast. But it works fine without it now.